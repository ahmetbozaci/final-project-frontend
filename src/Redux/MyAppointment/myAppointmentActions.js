import axios from 'axios';

export const FETCH_MY_APP_REQUEST = 'FETCH_MY_APP_REQUEST';
export const FETCH_MY_APP_SUCCESS = 'FETCH_MY_APP_SUCCESS';
export const FETCH_MY_APP_FAILURE = 'FETCH_MY_APP_FAILURE';
export const DELETE_MY_APP_SUCCESS = 'FETCH_MY_APP_FAILURE';

const baseURL = `${process.env.REACT_APP_DOCTOR_APPOINTMENT_API_URL}/appointments`;

const fetchMyAppRequest = () => ({
  type: FETCH_MY_APP_REQUEST,
});

const fetchMyAppSuccess = (myapp) => ({
  type: FETCH_MY_APP_SUCCESS,
  payload: myapp,
});

const fetchMyAppFailure = (myapp) => ({
  type: FETCH_MY_APP_FAILURE,
  payload: myapp,
});

export const deleteMyAppSuccess = (myapp) => ({
  type: DELETE_MY_APP_SUCCESS,
  payload: myapp,
});

export const deleteAppointment = (id) => async (dispatch) => {
  await fetch(`{baseURL}/${id}`, {
    method: 'DELETE',
  });
  dispatch(deleteMyAppSuccess(id));
};

export const fetchMyApp = () => (dispatch) => {
  dispatch(fetchMyAppRequest());
  axios.get(baseURL)
    .then((response) => {
      const myApp = response.data;
      dispatch(fetchMyAppSuccess(myApp));
    })
    .catch((error) => {
      const errorMsg = error.message;
      dispatch(fetchMyAppFailure(errorMsg));
    });
};
