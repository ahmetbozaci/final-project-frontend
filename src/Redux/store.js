import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './rootReducer';
import fetchDoctors from './doctor/doctorActions';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger)),
);
store.dispatch(fetchDoctors());

export default store;
