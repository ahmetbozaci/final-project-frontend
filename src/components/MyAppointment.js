import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Moment from 'react-moment';
import 'moment-timezone';
import { fetchMyApp } from '../Redux/MyApp/myAppActions';
import '../assets/style/myApp.css';
// import fetchDoctor from '../Redux/Doctor/doctorActions';

const MyAppointment = () => {
  const myAppData = useSelector((state) => state.myAppReducer.myApp);
  // const doctorData = useSelector((state) => state.doctorReducer.doctors);
  // console.log('doctor data inside appointment', doctorData);
  // const doctorId = doctorData.map((d) => d.id);
  // console.log('all doctors id', doctorId);

  // console.log('here is myAppData THE STATE', myAppData);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMyApp());
  }, []);

  // useEffect(() => {
  //   dispatch(fetchDoctor());
  // }, []);

  return (
    <div className="p-2 myAppMargin">
      <h1 className="p-2 text-center">My Appointments</h1>
      <div>
        {myAppData
                && myAppData.map((d) => (
                  <div className="card mb-3 p-2" key={d.id}>
                    <div className="row g-0">
                      <div className="col-md-4">
                        <h2>
                          Doctor name:
                          {' '}
                          {d.doctor_id}
                        </h2>
                      </div>
                      <div className="col-md-8">
                        <div className="card-body">
                          {/* <h5 className="card-title">Card title</h5> */}
                          <p className="card-text" style={{ fontSize: '1.5rem' }}>
                            Reservation Date:
                            {' '}
                            <small className="text-muted">{d.date}</small>
                          </p>
                          <p className="card-text" style={{ fontSize: '1.5rem' }}>
                            Reservation Time:
                            {' '}
                            <Moment format="HH:MM" className="text-muted">{d.time}</Moment>
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                ))}
      </div>
    </div>

  );
};

export default MyAppointment;
