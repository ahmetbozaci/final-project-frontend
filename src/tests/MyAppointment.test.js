import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import MyAppointment from '../components/MyAppointment';

it('MyAppointment renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><MyAppointment /></Router></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
