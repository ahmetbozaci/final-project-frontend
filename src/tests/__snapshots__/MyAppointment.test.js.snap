// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAppointment renders correctly 1`] = `
<div
  className="p-2 myAppMargin"
>
  <h1
    className="p-2 text-center"
  >
    My Appointments
  </h1>
  <h4
    className="p-3"
  >
    Currently you do not have any appointments
  </h4>
</div>
`;
